### PR TITLE
Fix plot with datetime labels

### PR DESCRIPTION
**Pull request by SimonHeybrock**

### CODE REVIEW

**Comment by nvaytet** [comment]
So with the plot example, 
```Py
da = make_dense_data_array(ndim=1)
da.coords['time'] = sc.epoch(unit='ns') + sc.arange('xx', da.sizes['xx'], unit='ns')
sc.plot(da, labels={"xx": "time"})
```
i get
Screenshot at 2022-02-16 11-08-13

but then if i make the underlying coord stretch out quadratically
```Py
da.coords['xx'].values = da.coords['xx'].values**2
```
I would have thought that the horizontal spacing between the dots should get larger towards the right hand side, but the numbers on the tick labels should still be from 0 to 50ns (i.e not change).
However, they do change, they go from 50ns to 2500ns:
Screenshot at 2022-02-16 11-10-29

**Comment by SimonHeybrock** [comment]
That seems like an unrelated bug?

**Comment by nvaytet** [comment]
> That seems like an unrelated bug?

It's not completely unrelated, we're just trying to get the plotting of datetime workign with labels?

File: src/scipp/plotting/formatters.py [hunk 1]
```diff
@@ -50,13 +50,15 @@ class DateFormatter:
...
+    def __init__(self, coord, offset, dim):
+        self.coord_values = coord.values
         self.offset = offset
         self.dim = dim
         self.indicators = []
 
     def formatter(self, val, pos, axis=None, get_axis_bounds=None, set_axis_label=None):
-        d = (self.offset + (int(val) * self.offset.unit)).value
+        index = np.abs(self.coord_values - val).argmin()
+        d = (self.offset + (index * self.offset.unit)).value
```
Comment (nvaytet) [inline src/scipp/plotting/formatters.py:61]: So all you need is `index`, and not something like `self.label_values[index]`?

Comment (SimonHeybrock) [reply src/scipp/plotting/formatters.py:61]: Well, as far as I understand the existing code, for datetime the labels are not looked up but *computed*?

Comment (nvaytet) [reply src/scipp/plotting/formatters.py:61]: i think it's fine.

**Comment by SimonHeybrock** [comment]
Not sure my fix is correct. There is an exception when you zoom in too much. I don't know if this also happened previously?

**Comment by nvaytet** [comment]
> Not sure my fix is correct. There is an exception when you zoom in too much. I don't know if this also happened previously?

I think this has always been there, I think @jl-wynen also complained about this in the past.
And the resoning was that once you zoom in this far, the plot is no longer so meaningful, you only have one or no points in the viewport?

**Comment by SimonHeybrock** [comment]
> > Not sure my fix is correct. There is an exception when you zoom in too much. I don't know if this also happened previously?
> 
> I think this has always been there, I think @jl-wynen also complained about this in the past. And the resoning was that once you zoom in this far, the plot is no longer so meaningful, you only have one or no points in the viewport?

Yes it has been there before, but it causes an exception and the mpl axes are not shown anymore.

**Comment by SimonHeybrock** [comment]
But let us leave that fix for another time...

**Comment by nvaytet** [comment]
hmm we broke the python tests...

File: src/scipp/plotting/formatters.py [hunk 2]
```diff
@@ -225,6 +225,7 @@ def make_formatter(array, key, dim):
             formatter["custom_locator"] = True
         elif kind == Kind.datetime:
             coord = _get_or_make_coord(array, dim)
+            coord = coord - coord.min()
```
Comment (nvaytet) [inline src/scipp/plotting/formatters.py:228]: I feel (also when I made the changes in #2446) a lot of this is duplicated between `formatter.py` and `model.py`.

**Review by nvaytet — APPROVED** [review]

**Pull request merged**